benchmarks/pipe.py

```diff
@@ -409,38 +409,38 @@ def generate_balance(num_devices, num_layers):
     return balance
 
 
-def get_synthetic_dataloader(args):
+def get_synthetic_dataloader(args, benchmark_config):
     """Returns dataloader for synthetic data."""
 
     if args.model_name == "lm":
-        return get_synthetic_wikitext2_dataloaders(args)
+        return get_synthetic_wikitext2_dataloaders(args, benchmark_config)
     else:
         raise RuntimeError("Unrecognized args.model_mame " % args.model_name)
 
 
-def get_real_dataloaders(args, device, config):
+def get_real_dataloaders(args, device, benchmark_config):
     """Returns dataloaders for real data."""
 
     if args.model_name == "lm":
-        data = get_real_wikitext2_dataloaders(args)
+        data = get_real_wikitext2_dataloaders(args, benchmark_config)
         ntokens, train_dataloader, valid_dataloader, test_dataloader = data
-        config["vocab_size"] = ntokens
+        benchmark_config["vocab_size"] = ntokens
         return train_dataloader, valid_dataloader, test_dataloader
     else:
         raise RuntimeError("Unrecognized args.model_mame " % args.model_name)
 
 
-def create_model_config(args, config=None):
+def create_model_config(args, benchmark_config=None):
     """Return a dict with the given model, dataset and optimizer."""
 
     device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
     if args.use_synthetic_data:
```

Comment:
if between if and else case the only difference is the function, I would do this:

if blah:
  func = f1
else:
  func = f2
common code
code that use func

this should reduce duplication.